test/node/tasks/StreamUpload.ts

```diff
@@ -51,3 +52,47 @@ describe("StreamUpload", () => {
 		assert.equal(sliceSize, (slice as Buffer).length);
 	});
 });
+
+describe("Stream upload resume", () => {
```

Comment:
Do these tests cover scenarios where:
- upload() fails before the bytes are uploaded so resume() uses nextExpectedRange = [0 , rangesize-1] to set nextRange.
- upload() fails after the bytes are uploaded so resume() uses nextExpectedRange = [n , n+rangesize-1] to set nextRange. This assumes we get a response.

Reply:
I think these cases are the same as resume() always use the uploadUrl from the session to get the nextExpectedRange.